pkg/node/node.go

```diff
@@ -424,41 +429,109 @@ func (n *Node) publishSignature(
 			continue
 		}
 
+		if !n.confirmSignature(keepAddress, digest) {
```

Comment:
I think we should perhaps add a blocking function `waitForSignature` that will block for e.g. 10 minutes and check every one minute if the signature has been submitted:
```
if !(n.waitForSignature(keepAddress, digest) && n.confirmSignature(keepAddress.digest))
```

The scenario I was testing:
Three group members submitted signature transactions at the same time. None of them has failed on gas estimation and all three transactions landed in the mempool. They were underpriced compared to other transactions and it took 10 blocks to mine first of them. In this scenario, `confirmSignature` verified only 2 confirmations.

It seems to me that this check is the only one affected. WDYT?

Reply:
Yep, you're right. I'll add the `waitForSignature` function.